Create CI

ilkerhalil opened the pull request.

ilkerhalil committed:
Create CI

change jobs name for cd wf

Added telemetry and nologo environment for dotnet setup

Update .github/workflows/cd.yml

use DAEMON_NUGET_PUBLIC in all cases

change Nuget secrets, Added telemetry and nologo environment for dotnet setup

Rjvs commented on .github/workflows/cd.yml:
```diff
@@ -0,0 +1,118 @@
+    tags:
+      - 'v*'
+jobs:
+  build_and_upload_artifact_for_macos_and_linux:
```
```suggestion
  build_and_upload:
```

Rjvs commented on .github/workflows/cd.yml:
```diff
@@ -0,0 +1,118 @@
+        with:
+          name: bufc-${{ matrix.config.name }}
+          path: cmake-build-release/bufc
+  build_and_upload_artifact_for_windows:
```
```suggestion
  build_and_upload_win:
```
The name is too long and I can't see the platform

Rjvs commented on .github/workflows/cd.yml:
```diff
@@ -0,0 +1,118 @@
+          submodules: true
+      - name: Fetch tag git metadata
+        run: git fetch --force --update-shallow --depth 1 origin 'refs/tags/*:refs/tags/*'
+      - uses: actions/setup-dotnet@v1.8.1
+        with:
+          dotnet-version: 5.0.x
```
```suggestion
      - uses: actions/setup-dotnet@v1.8.1
        env:
          DOTNET_NOLOGO: true
          DOTNET_CLI_TELEMETRY_OPTOUT: true
        with:
          dotnet-version: 5.0.x
```

Rjvs commented on .github/workflows/cd.yml:
```diff
@@ -0,0 +1,118 @@
+      - name: NuGet Auth
+        uses: StirlingLabs/GithubNugetAuthAction@main
+        with:
+          token: ${{ secrets.DAEMON_NUGET }}
```
```suggestion
          token: ${{ secrets.DAEMON_NUGET_PUBLIC }}
```

TYoungSL committed:
troubleshoot token permissions

TYoungSL committed:
try to fix release config

TYoungSL committed:
rename Version.Net -> StirlingLabs.Version

add version metadata project imports

add GitHubActionsTestLogger

add Configuration folder to solution

TYoungSL committed:
clean up debugging usages of pwd

specify solution now that there's version.proj in the directory
